### ISSUE TITLE
Odd CPU usage at idle

### ISSUE THREAD

**Issue by TunaLatte** [KevinOConnor/klipper#17]
Hello,

I have noticed that whenever klippy is idle disconnected or connected and not printing or doing anything (e.g if I set the extruder temp cpu goes down to 1%, when temperature is reach, spikes 100 again and alters between 100 and 1 in slow frequency, if i turn it off, its back at 100% all the time), cpu spikes to 100% on one cpu thread:
![screenshot_20170614_225151](https://user-images.githubusercontent.com/24957562/27151783-5f7c1a6c-5154-11e7-8314-3473bcd55fe0.png)

Also CPU spikes immediatly after repeating pattern of the 'no serial found' message:
![image](https://user-images.githubusercontent.com/24957562/27151966-fa2a23a6-5154-11e7-90fd-99abda03103c.png)

Would be great to hear feedback from other users to isolate whether its an issue with my config/environment or klippy bug. Running on desktop, python version: Python 2.7.12 
```
tmash@tmash-nuc:~$ pip freeze
beautifulsoup4==4.4.1
chardet==2.3.0
cycler==0.9.0
html5lib==0.999
iotop==0.6
lxml==3.5.0
matplotlib==1.5.1
numpy==1.11.0
Pillow==3.1.2
Pivy==0.5.0
ply==3.7
pycollada==0.4
pygobject==3.20.0
PyOpenGL==3.0.2
pyparsing==2.0.3
pyserial==3.0.1
python-dateutil==2.4.2
pytz==2014.10
six==1.10.0
smbus==1.1
vboxapi==1.0
virtualenv==15.0.1
wxPython==3.0.2.0
wxPython-common==3.0.2.0
youtube-dl==2016.2.22

```
EDIT1: To add, just tried the python3 branch, same issue.

**Comment by KevinOConnor** [comment]
On Wed, Jun 14, 2017 at 02:04:35PM -0700, TunaLatte wrote:
> I have noticed that whenever klippy is idle disconnected or connected and not printing or doing anything (e.g if I set the extruder temp cpu goes down to 1%, when temperature is reach, spikes 100 again and alters between 100 and 1 in slow frequency, if i turn it off, 100% all the time), cpu spikes to 100% on one cpu thread:

Thanks.  I think that should be fixed now (as of commit c03b4921).

FYI, separately, I'd recommend running the Klipper host software as a
non-root user.

-Kevin

**Comment by TunaLatte** [comment]
Thanks! CPU average is down to 1% average.
And yeah both octoprint and klippy were running as root as I was diagnosing something with socat permissions for wireless printing.
Issue resolved.
